### PR TITLE
nss: drop multiple patches

### PR DESCRIPTION
**Pull request by mweinelt**
###### Things done

Removed a bunch of patches that I don't believe serve a purpose any more.

- nss-pem patchset went 404 and if people still want that functionality they/we should patch nss-pem
- The ckpem patch was added for darwin compat, waiting for ofBorg to see if that is necessary
- Quickly tested `pkgsCross.raspberryPi.nss` without the cross-compilation patch, built just fine
  - but pkgsCross.armv7l-hf-multiplatform.nss fails without the patch, commit dropped.

Based on master for easier testing, but will go into staging, once people are happy with it.

Open question: What were the consumer of the nss-pem patch? Should we package nss-pem in this PR?

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ajs124 — APPROVED** [review]
diff LGTM. this is probably fine, if darwin still works.

**Comment by mweinelt** [comment]
@ofborg build nss nss_latest

**Comment by vcunat** [comment]
All four platforms succeeded on OfBorg.  (Before rebasing.  It's not easy to see anymore, so let me state it.)

File: pkgs/development/libraries/nss/generic.nix [hunk 1]
```diff
@@ -45,13 +40,6 @@ stdenv.mkDerivation rec {
   propagatedBuildInputs = [ nspr ];
 
   prePatch = ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nss/generic.nix:42]: This should really be moved all to postPatch otherwise some patches might not apply.

**Review by ajs124 — APPROVED** [review]

**Pull request merged**